test/Parser.test.js

```diff
@@ -227,9 +227,9 @@ describe("Parser", function() {
 			"typeof 'str'": "string=string",
 			"typeof aString": "string=string",
 			"typeof b.Number": "string=number",
-			"typeof b[Number]": "",
+			"typeof b[Number]": "string=number",
```

Comment:
this is incorrect. It should handle `b["Number"]` not `b[Number]`. add explicit testcases for this new syntax